### PR TITLE
 [FLINK-13384][runtime] Fix back pressure sampling for SourceStreamTask

### PR DESCRIPTION
**Pull request by GJL**
See #9271

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTask.java [hunk 1]
```diff
@@ -58,6 +61,7 @@
 
 	public SourceStreamTask(Environment env) {
 		super(env);
+		this.sourceThread = new LegacySourceFunctionThread(getName());
```
Comment (GJL) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTask.java]: Here we are calling a non-final method from the constructor :(

Comment (GJL) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTask.java]: The name could be set in `performDefaultAction()`, before `sourceThread.start();`

Comment (pnowojski) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTask.java]: Not the best solution, but let's not overthink this. +1 for setting the name in `performDefaultAction`

**Comment by GJL** [comment]
Merged

**Pull request closed without merging**